service/credential_adapter/mock/mock_credential_adapter.go

```diff
@@ -65,18 +65,18 @@ func (mr *MockCredentialAdapterClientMockRecorder) CreateIssuanceChallenge(cred_
 }
 
 // CreatePresentationChallenge mocks base method.
-func (m *MockCredentialAdapterClient) CreatePresentationChallenge() (*credential_adapter.PresentationChallenge, error) {
+func (m *MockCredentialAdapterClient) CreatePresentationChallenge(credTypes []string) (*credential_adapter.PresentationChallenge, error) {
 	m.ctrl.T.Helper()
-	ret := m.ctrl.Call(m, "CreatePresentationChallenge")
+	ret := m.ctrl.Call(m, "CreatePresentationChallenge", credTypes)
 	ret0, _ := ret[0].(*credential_adapter.PresentationChallenge)
 	ret1, _ := ret[1].(error)
 	return ret0, ret1
 }
 
 // CreatePresentationChallenge indicates an expected call of CreatePresentationChallenge.
-func (mr *MockCredentialAdapterClientMockRecorder) CreatePresentationChallenge() *gomock.Call {
+func (mr *MockCredentialAdapterClientMockRecorder) CreatePresentationChallenge(credTypes interface{}) *gomock.Call {
```

Comment:
Is it possible to avoid using interface{} and cast it to certain type or interface?

Reply:
This code is automatically generated by gomock, I have no control over it to that level. Also in order for the mocking to work it would have to be `interface{}`.

Btw in general, the mock_{whatever}.go files are never edited and always automatically generated. So it's safe to ignore them during reviews.

Reply:
But I do agree with you that whenever possible we should absolutely avoid `interface{}`